crates/uv-interpreter/src/interpreter.rs

```diff
@@ -49,41 +51,56 @@ impl Interpreter {
         );
 
         Ok(Self {
-            platform: PythonPlatform(platform.to_owned()),
+            platform,
             markers: Box::new(info.markers),
+            sysconfig: info.sysconfig,
+            prefix: info.prefix,
             base_exec_prefix: info.base_exec_prefix,
             base_prefix: info.base_prefix,
-            stdlib: info.stdlib,
             sys_executable: info.sys_executable,
             tags: OnceCell::new(),
         })
     }
 
     // TODO(konstin): Find a better way mocking the fields
-    pub fn artificial(
-        platform: Platform,
-        markers: MarkerEnvironment,
-        base_exec_prefix: PathBuf,
-        base_prefix: PathBuf,
-        sys_executable: PathBuf,
-        stdlib: PathBuf,
-    ) -> Self {
+    pub fn artificial(platform: Platform, markers: MarkerEnvironment) -> Self {
         Self {
-            platform: PythonPlatform(platform),
+            platform,
             markers: Box::new(markers),
-            base_exec_prefix,
-            base_prefix,
-            stdlib,
-            sys_executable,
+            sysconfig: Sysconfig {
+                purelib: PathBuf::from("/dev/null"),
+                platlib: PathBuf::from("/dev/null"),
+                scripts: PathBuf::from("/dev/null"),
+                include: PathBuf::from("/dev/null"),
+                data: PathBuf::from("/dev/null"),
+            },
+            prefix: PathBuf::from("/dev/null"),
+            base_exec_prefix: PathBuf::from("/dev/null"),
+            base_prefix: PathBuf::from("/dev/null"),
+            sys_executable: PathBuf::from("/dev/null"),
             tags: OnceCell::new(),
         }
     }
 
-    /// Return a new [`Interpreter`] with the given base prefix.
+    /// Return a new [`Interpreter`] with the given virtual environment root.
     #[must_use]
-    pub(crate) fn with_base_prefix(self, base_prefix: PathBuf) -> Self {
+    pub(crate) fn with_venv_root(self, venv_root: PathBuf) -> Self {
+        let platform = PythonPlatform::from(self.platform.clone());
         Self {
-            base_prefix,
+            // Given that we know `venv_root` is a virtualenv, and not an arbitrary Python
+            // interpreter, we can safely assume that the platform is the same as the host
+            // platform. Further, we can safely assume that the paths follow a predictable
+            // structure, which allows us to avoid querying the interpreter for the `sysconfig`
+            // paths.
+            sysconfig: Sysconfig {
+                purelib: platform.venv_site_packages(&venv_root, self.python_tuple()),
+                platlib: platform.venv_site_packages(&venv_root, self.python_tuple()),
+                scripts: platform.venv_scripts_dir(&venv_root),
+                data: platform.venv_data_dir(&venv_root),
+                include: self.sysconfig.include,
```

Comment:
\cc @konstin - I think `include` is the same in a venv as not...

Reply:
Odd, the docs make it look as if should also change.

Reply:
https://docs.python.org/3/library/sysconfig.html#installation-paths

But i don't disagree, `include` is also an outlier for me:

```
Paths: 
        data = "/home/konsti/projects/uv/.venv"
        include = "/home/konsti/.pyenv/versions/3.10.13/include/python3.10"
        platinclude = "/home/konsti/.pyenv/versions/3.10.13/include/python3.10"
        platlib = "/home/konsti/projects/uv/.venv/lib/python3.10/site-packages"
        platstdlib = "/home/konsti/projects/uv/.venv/lib/python3.10"
        purelib = "/home/konsti/projects/uv/.venv/lib/python3.10/site-packages"
        scripts = "/home/konsti/projects/uv/.venv/bin"
        stdlib = "/home/konsti/.pyenv/versions/3.10.13/lib/python3.10"
```

Same with the apt installed python as base:

```
Paths: 
        data = "/home/konsti/projects/uv/.venv"
        include = "/usr/include/python3.11"
        platinclude = "/usr/include/python3.11"
        platlib = "/home/konsti/projects/uv/.venv/lib/python3.11/site-packages"
        platstdlib = "/home/konsti/projects/uv/.venv/lib/python3.11"
        purelib = "/home/konsti/projects/uv/.venv/lib/python3.11/site-packages"
        scripts = "/home/konsti/projects/uv/.venv/bin"
        stdlib = "/usr/lib/python3.11"
```

Reply:
Depends on the build headers I think? There was an issue a while ago with deadsnakes changing them to `/usr/local/include`  vs canonical `/usr/include` https://github.com/deadsnakes/issues/issues/237